src/components/Treeview/TreeviewItem.tsx

```diff
@@ -13,27 +13,47 @@ export class TreeviewItem extends CommonComponent<ITreeviewItemProps, any> {
         isOpen: false
     };
 
-    constructor(props) {
+    private readonly expandedIcon: string = 'icon-arrow_down_right';
+    private readonly collapsedIcon: string = 'icon-arrow_right';
+
+    private readonly onExpand;
+    private readonly getIsOpen;
+
+    public constructor(props: ITreeviewItemProps) {
         super(props);
+
         this.state = {
-            isOpen: props.isOpen,
-            iconArrow: 'icon-arrow_right',
-            hover: false
+            hover: false,
+            isOpen: props.onExpand !== undefined ? undefined : props.item.isOpen
+        };
+
+        this.onExpand = props.onExpand !== undefined ? props.onExpand : (itemId: string, expanded: boolean) => {
+            let { isOpen } = this.state;
+            this.setState({
+                isOpen: expanded
+            });
         };
-    }
 
+        this.getIsOpen = props.onExpand !== undefined ?
+            () => this.props.item.isOpen !== undefined ? this.props.item.isOpen : false :
+            () => this.state.isOpen;
+    }
 
     public render(): JSX.Element {
-        let { item, onChange, showCheckbox, children, recursive } = this.props;
-        let { isOpen } = this.state;
+        let { item, onChange, showCheckbox, children, recursive, expandParentOnClick, onExpand } = this.props;
         let checkedStatus = this._getChildrenChecked(item, item.checked, recursive);
         let checked = checkedStatus.isChecked;
+
+        const isOpen = this.getIsOpen();
+        const arrowIcon = this.getArrowIconFromState(isOpen);
+
         const itemClassName = classNames(
```

Comment:
const itemClassName =  this.state.isOpen ? 'expanded' :  'collapsed';
the same for other class names that don't have multiple options